src/SupabaseClient.ts

```diff
@@ -92,18 +99,22 @@ export default class SupabaseClient<
     const defaultStorageKey = `sb-${new URL(this.authUrl).hostname.split('.')[0]}-auth-token`
     this.storageKey = options?.auth?.storageKey ?? defaultStorageKey
 
-    const settings = { ...DEFAULT_OPTIONS, ...options, storageKey: this.storageKey }
+    const settings = Object.assign(DEFAULT_OPTIONS, options)
```

Comment:
I just have those concerns:

1. If `options.auth.storageKey` is not set, the `defaultStorageKey` seems to not be applied
2. By using `Object.assign()` `DEFAULT_OPTIONS` gets changed, if that should not be the case it is safer to use `Object.assign({}, DEFAULT_OPTIONS, options)`  which creates a new object and assigns to it.
3. `Object.assign()` only does a shallow merge, eg if `options.auth` is set it will completely replace `DEFAULT_OPTIONS.auth`

Reply:
Hey @pixtron,

Thanks for raising these point!  You are right, the previous draft implementation which makes use of `Object.assign` doesn't work  as `Object.assign` does a shallow copy. This was flagged two days backbit we hadn't gotten around to fixing it.

I've made some amendments by drawing heavily on the changes you've suggested in the branch and will tag you for review once I clean up and write tests.

Once again, thanks so much for the feedback!